### PR TITLE
Accept relayed tx v3 with sender account non-existent

### PR DESCRIPTION
**Pull request by sstanculeanu**
## Reasoning behind the pull request
- relayed txs v3 with sender account not-existent are rejected by interceptors, despite the zero value of transaction
  
## Proposed changes
- allow relayed txs v3 with new senders, in case of zero value transactions

## Testing procedure
- with feat branch

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: process/dataValidators/txValidator.go [hunk 1]
```diff
@@ -76,14 +77,27 @@ func (txv *txValidator) CheckTxValidity(interceptedTx process.InterceptedTransac
 		return nil
 	}
 
+	// for relayed v3, we allow sender accounts that do not exist
+	isRelayedV3 := common.IsRelayedTxV3(interceptedTx.Transaction())
+	hasValue := hasTxValue(interceptedTx)
+	shouldAllowMissingSenderAccount := isRelayedV3 && !hasValue
 	accountHandler, err := txv.getSenderAccount(interceptedTx)
-	if err != nil {
+	if err != nil && !shouldAllowMissingSenderAccount {
```
Comment (andreibancioiu) [inline process/dataValidators/txValidator.go:85]: Should we check for `ErrAccountNotFound` (explicitly)? Just a question.

File: process/dataValidators/txValidator.go [hunk 2]
```diff
@@ -76,14 +77,27 @@ func (txv *txValidator) CheckTxValidity(interceptedTx process.InterceptedTransac
...
 		return err
 	}
 
 	return txv.checkAccount(interceptedTx, accountHandler)
 }
 
+func hasTxValue(interceptedTx process.InterceptedTransactionHandler) bool {
```
Comment (andreibancioiu) [inline process/dataValidators/txValidator.go:92]: Indeed, we don't handle `MultiESDTTransfer` with EGLD in interceptor (we never did, but that is OK).

File: process/dataValidators/txValidator.go [hunk 3]
```diff
@@ -76,14 +77,27 @@ func (txv *txValidator) CheckTxValidity(interceptedTx process.InterceptedTransac
...
+	txValue := interceptedTx.Transaction().GetValue()
+	if check.IfNilReflect(txValue) {
+		return false
+	}
+
+	return big.NewInt(0).Cmp(txValue) < 0
```
Comment (andreibancioiu) [inline process/dataValidators/txValidator.go:98]: Maybe directly use `Sign()` method of `txValue`, without instantiating a new big int (zero)?

**Review by andreibancioiu — APPROVED** [review]

File: process/dataValidators/txValidator.go (hunk 1 above)
Comment (sstanculeanu) [reply process/dataValidators/txValidator.go:85]: I don't think it is necessary, as `ErrAccountNotFound` is the only one returned from `getSenderAccount` method

File: process/dataValidators/txValidator.go (hunk 3 above)
Comment (sstanculeanu) [reply process/dataValidators/txValidator.go]: pushed

**Review by AdoAdoAdo — APPROVED** [review]

**Review by andreibancioiu — APPROVED** [review]

**Pull request merged**